src/features/foldingProvider.ts

```diff
@@ -99,16 +99,43 @@ export default class AsciidocFoldingRangeProvider implements vscode.FoldingRange
     }
   }
 
+  private static handleSingleLineCommentFoldingRanges (singleLineCommentStartIndexes: any[], foldingRanges: any[], lineIndex: number, lineText: string,
+    isPreviousLineSingleCommented: boolean) {
+    if (singleLineCommentStartIndexes.length === 0) {
+      isPreviousLineSingleCommented = false
+    }
+    if (singleLineCommentStartIndexes.length !== 0) {
+      isPreviousLineSingleCommented = true
+    }
+    if (lineText.slice(0, 2) === '//' && !isPreviousLineSingleCommented) {
+      singleLineCommentStartIndexes.push(lineIndex)
+      isPreviousLineSingleCommented = true
+    }
+    if (lineText.slice(0, 2) !== '//' && isPreviousLineSingleCommented) {
+      const startIndex = singleLineCommentStartIndexes.pop()
+      const endIndex = lineIndex - 1
+      if (endIndex - startIndex > 0) {
+        foldingRanges.push(new vscode.FoldingRange(
```

Comment:
We should also create a folding range when `singleLineCommentStartIndexes` is not empty and we are on the last line. For instance:

```
This is a paragraph.

// This is a comment.
// The last line of the document is also a comment!
```